graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/execution/ContextWebFilter.kt

```diff
@@ -32,14 +33,23 @@ const val GRAPHQL_CONTEXT_FILTER_ODER = 0
 /**
  * Default web filter that populates GraphQL context in the reactor subscriber context.
  */
-class ContextWebFilter(private val contextFactory: GraphQLContextFactory<Any>) : WebFilter, Ordered {
+class ContextWebFilter(config: GraphQLConfigurationProperties, private val contextFactory: GraphQLContextFactory<Any>) : WebFilter, Ordered {
+    private val contextualPathsRegex = "/${config.endpoint}|/${config.subscriptions.endpoint}".toRegex(RegexOption.IGNORE_CASE)
 
     @Suppress("ForbiddenVoid")
-    override fun filter(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Void> = mono {
-        contextFactory.generateContext(exchange.request, exchange.response)
-    }.flatMap { graphQLContext ->
-        chain.filter(exchange).subscriberContext { it.put(GRAPHQL_CONTEXT_KEY, graphQLContext) }
-    }
+    override fun filter(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Void> =
+        if (isApplicable(exchange)) {
+            mono {
+                contextFactory.generateContext(exchange.request, exchange.response)
+            }.flatMap { graphQLContext ->
+                chain.filter(exchange).subscriberContext { it.put(GRAPHQL_CONTEXT_KEY, graphQLContext) }
+            }
+        } else {
+            chain.filter(exchange)
+        }
 
     override fun getOrder(): Int = GRAPHQL_CONTEXT_FILTER_ODER
+
+    internal fun isApplicable(exchange: ServerWebExchange): Boolean =
+        contextualPathsRegex.matches(exchange.request.uri.path)
```

Comment:
what's the impact of this regex match on every request?

Reply:
should be negligible. I could change it to string equality check unsure whether it will be any different.

Reply:
think this should be fine.

Reply:
updated to string comparison as it seemed to be slightly faster in local testing (100_000 executions took ~25ms)